llvm/test/CMakeLists.txt

```diff
@@ -59,100 +59,106 @@ configure_lit_site_cfg(
 
 # Set the depends list as a variable so that it can grow conditionally.
 # NOTE: Sync the substitutions in test/lit.cfg when adding to this list.
+
+set(LLVM_TEST_DEPENDS_COMMON
+  FileCheck
+  count
+  llvm-config
+  not
+  prepare-check-lit
```

Comment:
Why is `prepare-check-lit` here? This shouldn't be needed by anything but `llvm/utils/lit` tests.

It breaks the build for us, since we run `lit` tests separately, since they were flaky in the past.

Reply:
See my comment above:
>  Additionally, I found that if I rm -rf build/bin and then do ninja -C build check-llvm-filecheck it would fail due to missing llvm-lit. So added prepare-check-lit to LLVM_TEST_DEPENDS_COMMON to resolve it.

But maybe the right thing is to add the prepare-check-lit to the depends for these specialized test suites,

Reply:
Can you check if https://github.com/llvm/llvm-project/pull/159922 fixes your issue?

Reply:
I also started seeing "Preparing lit tests" message that I haven't seen before. (It doesn't break build for me, but the copying requires some time, which can be frustrating.)

Reply:
`bin/llvm-lit` is generated by `configure_file`, which will only "reconfigure" the file again if the input file has changed.
It will not "reconfigure" the generated file if it was deleted/modified, for that to happen you need to re-run CMake.